### DISCUSSION TITLE
Searxng on Debian 12

### DISCUSSION THREAD

**Discussion by Shaggy1444** [searxng/searxng#3845] [Q&A] [answered]
Ran into issue upgrading current searxng install to Debian 12.

After reboot, not able to access searxng via Browser (Brave).

-OS Upgrade to Debian-bookworm causes internal server error when loading searxng web application
		-ModuleNotFound error: No module named 'pip'

After digging, found that Debian 12 upgrade process removed Python and PIP and who knows whatever else.

Restored back to Debian 11-latest searxng install.

So the questions are:

is searxng supported on Debian 12?
if it is, is it best to just run the install script as per documention?

Any help appreciated - thanks in advance!

**Comment by Fijxu** [comment]
Yes, it is supported. I think the problem is that in the update, SearxNG keep using the old pip that doesn't exist anymore after the update. Do you use UWSGI right? If so, then just try reinstalling.

I might be wrong, but a reinstall would fix it anyways. You don't need to do it from zero tho.

**Reply by Shaggy1444** [reply, marked as answer]
This is what I ended-up doing:

1.	update then upgrade current debian with all current updates and upgrades
2.	uninstall all searxng components
3.	change debian apt libraries from current debian version to new version
4.	update then upgrade new debian version with all current updates and upgrades
5.	if 4 is successful, autoremove and clean defunct packages
		-do a server backup then restart server
6.	sign onto already existing searx user id 
7.	get new/current searxng components
8.	install new searxng components and dependencies
9.	test to verify whether new searxng is working and accessible by browser

-then it fails, but this time with:
502 Bad Gateway
nginx/1.22.1

-checked and found that
/etc/uwsgi/apps-available/searxng.ini 
is missing although the link to it from
/etc/uwsgi/apps-enabled/searxng.ini is there

-so what happened to the searxng.ini during the install?  beats me!
-the documentation is confusing
-so I'm stuck perhaps at configuring the uwsgi component since nginx is dishing-out the 502

Quote:  things that should be simple are hard

**Comment by Shaggy1444** [comment]
Seems I discovered my own answer!!  Yeah!!

-so I did a find searxng/ -name *.ini against the installation folder
-it found searxng/utils/templates/etc/uwsgi/apps-available/searxng.ini
-so I copied it to my local folder and edited it as per the uwsgi documentation - https://docs.searxng.org/admin/installation-uwsgi.html
-then copied it to /etc/uwsgi/apps-available/searxng.ini after validating changes I made
-then restarted the uwsgi service - sudo service uwsgi restart searxng
-browsed my site and did a search and everything working

Hope this helps someone.
Best!!

**Comment by return42** [comment]
> After digging, found that Debian 12 upgrade process removed Python and PIP and who knows whatever else.

- Related: https://github.com/searxng/searxng/issues/3235
- Fixed in: https://github.com/searxng/searxng/pull/3748
